sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -50,6 +50,8 @@ trait InvokeLike extends Expression with NonSQLExpression with ImplicitCastInput
 
   def propagateNull: Boolean
 
+  override def foldable: Boolean =
+    children.forall(_.foldable) && !dataType.isInstanceOf[ObjectType]
```

Comment:
quick question: why do we exclude `ObjectType`?

Reply:
> quick question: why do we exclude `ObjectType`?

Since current Literal.create() not support ObjectType.  I will do this later

Reply:
Hm, then we can just remove `&& !dataType.isInstanceOf[ObjectType]` no?

Reply:
> Hm, then we can just remove `&& !dataType.isInstanceOf[ObjectType]` no?

Hmmm, you mean we can just remove `&& !dataType.isInstanceOf[ObjectType]` later after enable `Literal.create()` support ObjectType? Plan to do like this.

Reply:
I meant to just remove `ObjectType` since `ObjectType` is not supported in `Literal` anyway so this condition is uesless .. or did I miss something?

Reply:
> I meant to just remove `ObjectType` since `ObjectType` is not supported in `Literal` anyway so this condition is uesless .. or did I miss something?

Discussed with @cloud-fan , ObjectType only used in InvokeLike. WDYT @cloud-fan ?

Reply:
We must exclude `ObjectType` otherwise constant folding will fail when it tries to create a literal with `ObjectType`.